### PR TITLE
setting link destination in Project view w. hardcoded source page

### PR DESCRIPTION
**Pull request by gvn**

### CODE REVIEW

**Comment by gvn** [comment]
@thisandagain This is more progress toward #1670. I'm trying to keep PRs in smaller chunks...also I just had a rather nasty rebase. :rage1: 

So far this only works in browser and relies on a hardcoded value for the "source page".

I don't think landing this will interfere with any on device functionality. We may or may not want to take out my new router fallback params though...

File: www_src/pages/project/project.jsx [hunk 1]
```diff
@@ -278,6 +306,10 @@ var Project = React.createClass({
       state.camera = this.cartesian.getFocusTransform({x: 0, y: 0}, this.state.zoom);
     }
     this.setState(state);
+
+    if (this.state.params.mode === 'linkDestination') {
+      this.highlightPage('408', 'source'); // TEMP / TODO - Hardcoded. Should come from passed value.
```
Comment (gvn) [inline www_src/pages/project/project.jsx]: Note this hardcoded page ID...works as of now, but if the project loses this page you might need to set another ID to see this working...

File: www_src/lib/router.jsx [hunk 2]
```diff
@@ -20,6 +20,8 @@ module.exports = {
       }
     } else {
       params = {
+        // edit, play, linkDestination
+        mode: 'linkDestination',
```
Comment (gvn) [inline www_src/lib/router.jsx]: `edit` might be a more sensible default...?

Comment (thisandagain) [inline www_src/lib/router.jsx]: :+1:

File: www_src/pages/project/project.jsx [hunk 3]
```diff
@@ -208,11 +213,33 @@ var Project = React.createClass({
...
-    this.setState({
-      camera: this.cartesian.getFocusTransform(el.coords, this.state.zoom),
-      selectedEl: el.id
-    });
+  /**
+   * Highlight a page in the UI and move camera to center it
+   * @param  {Number|String} id ID of page
+   * @param  {Number|String} type Type of highlight ("selected", "source")
+   */
+  highlightPage: function (id, type) {
```
Comment (thisandagain) [inline www_src/pages/project/project.jsx]: Nice. I like moving this to a self-contained method.

File: www_src/pages/project/project.jsx [hunk 4]
```diff
@@ -278,6 +305,10 @@ var Project = React.createClass({
       state.camera = this.cartesian.getFocusTransform({x: 0, y: 0}, this.state.zoom);
     }
     this.setState(state);
+
+    if (this.state.params.mode === 'linkDestination') {
```
Comment (thisandagain) [inline www_src/pages/project/project.jsx]: This could use a line comment.

File: www_src/static/img/oval-droplet.svg [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<svg width="52" height="52" viewBox="0 0 52 52" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (thisandagain) [inline www_src/static/img/oval-droplet.svg]: Would be fine to put this through SVGO

File: www_src/static/img/oval-flag.svg [hunk 6]
```diff
@@ -0,0 +1,22 @@
+<svg width="52" height="52" viewBox="0 0 52 52" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (thisandagain) [inline www_src/static/img/oval-flag.svg]: Same here

**Comment by thisandagain** [comment]
This looks good. No major blockers from my perspective. Just a few small notes and then R+

File: www_src/static/img/oval-droplet.svg (hunk 5 above)
Comment (gvn) [inline www_src/static/img/oval-droplet.svg]: I actually did, just pretty printed. :poodle:

File: www_src/pages/project/project.jsx (hunk 3 above)
Comment (gvn) [inline www_src/pages/project/project.jsx]: Yeah, I tried it as 2 methods at first, which was more single responsibility-ish, but way less DRY...so...yeah. :cactus:

**Pull request merged**